src/sys/windows/named_pipe.rs

```diff
@@ -381,6 +382,13 @@ impl NamedPipe {
         }
     }
 
+    /// Set the buffer size.
+    ///
+    /// The default is *currently* 4kB (4096 bytes).
+    pub fn set_buffer_size(&mut self, capacity: usize) {
+        self.inner.default_buf_size.store(capacity, Release)
```

Comment:
You aren't using the atomic to synchronize memory.
```suggestion
        self.inner.default_buf_size.store(capacity, Relaxed)
```